src/vulkan/vk-command.cpp

```
@@ -342,6 +342,15 @@ void CommandRecorder::cmdCopyTextureToBuffer(const commands::CopyTextureToBuffer
     // adjustedExtent.height = math::calcAligned(adjustedExtent.height, formatInfo.blockHeight);
 
     // Calculate the row length (in texels) from the supplied pitch (in bytes)
+    if (dstRowPitch % formatInfo.blockSizeInBytes != 0)
+    {
```

Cppcheck (2.21.0)

[portability] Shifting signed 32-bit value by 31 bits is implementation-defined behaviour

(shiftTooManyBitsSigned)